staging/src/k8s.io/kubectl/pkg/cmd/create/create_cronjob_test.go

```diff
@@ -112,9 +118,279 @@ func TestCreateCronJob(t *testing.T) {
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			err := test.options.Validate()
+			if test.expectErr && err == nil {
+				t.Errorf("expected error but validation passed")
+			}
+			if !test.expectErr && err != nil {
+				t.Errorf("unexpected error: %v", err)
+			}
+		})
+	}
+}
+
+func TestCompleteCreateCronJob(t *testing.T) {
+	// TODO: Test that CreateCronJobOptions.Command is properly set.
+	//       When setting .Command, CompleteCreateCronJob relies heavily
+	//       on Cobra, which makes testing difficult.
+
+	defaultTestName := "my-cronjob"
+	defaultTestDryRun := true
+	defaultTestRestart := "Never"
+
+	tf := cmdtesting.NewTestFactory().WithNamespace("test")
+	defer tf.Cleanup()
+
+	tf.Client = &fake.RESTClient{}
+	tf.ClientConfigVal = cmdtesting.DefaultClientConfig()
+
+	tests := map[string]struct {
+		params    []string
+		dryRun    bool
+		options   *CreateCronJobOptions
+		expected  *CreateCronJobOptions
+		expectErr bool
+	}{
+		"test-missing-name": {
+			params:    []string{},
+			options:   &CreateCronJobOptions{},
+			expected:  &CreateCronJobOptions{},
+			expectErr: true,
+		},
+		"test-missing-restart": {
+			params: []string{defaultTestName},
+			options: &CreateCronJobOptions{
+				PrintFlags: genericclioptions.NewPrintFlags("created"),
+			},
+			expected: &CreateCronJobOptions{
+				Name:    defaultTestName,
+				Restart: "OnFailure",
+			},
+			expectErr: false,
+		},
+		"test-valid-complete-case": {
+			params: []string{defaultTestName},
+			dryRun: defaultTestDryRun,
+			options: &CreateCronJobOptions{
+				PrintFlags: genericclioptions.NewPrintFlags("created"),
+				Restart:    defaultTestRestart,
+			},
+			expected: &CreateCronJobOptions{
+				Name:    defaultTestName,
+				DryRun:  defaultTestDryRun,
+				Restart: defaultTestRestart,
+			},
+			expectErr: false,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			cmd := NewCmdCreateRole(tf, genericclioptions.NewTestIOStreamsDiscard())
+
+			if test.dryRun {
+				cmd.Flags().Set("dry-run", "true")
+			}
+
+			err := test.options.Complete(tf, cmd, test.params)
+
+			if test.expectErr && err == nil {
+				t.Errorf("expected error but none was returned")
+			}
+			if !test.expectErr && err != nil {
+				t.Errorf("unexpected error: %v", err)
+			}
+
+			fields := map[string]struct {
+				expected, actual interface{}
+			}{
+				".Name": {
+					expected: test.expected.Name,
+					actual:   test.options.Name,
+				},
+				".DryRun": {
+					expected: test.expected.DryRun,
+					actual:   test.options.DryRun,
+				},
+				".Restart": {
+					expected: test.expected.Restart,
+					actual:   test.options.Restart,
+				},
+			}
+
+			for name, value := range fields {
+				if !reflect.DeepEqual(value.expected, value.actual) {
+					t.Errorf("mismatched field %q:\n%s", name, diff.ObjectReflectDiff(value.expected, value.actual))
+				}
+			}
+		})
+	}
+}
+
+func TestRunCreateCronJob(t *testing.T) {
```

Comment:
This is mostly duplicating `TestCreateCronJob`, if you look at what `Run` in `create_cronjob.go` does it's only creating the CJ itself and I don't think we need that much of testing of the printing and all the rest of the machinery. This test has more setup code than actual code, and one of the reasons we've went with `*Options` struct was to get rid of those additional setup steps. The high level testing should happen in test-cmd, instead.